README.md

```diff
@@ -303,23 +362,228 @@ jobs:
+
+## Passing env vars to reusable workflows
+
+If you want to use environment variables, passing them to a reusable workflow is 
+a little tricky as this behavior is not currently supported by GitHub.
+This is a [requested feature](https://github.community/t/passing-environment-variables-to-reusable-workflow/230456/4).
+The current workaround for this is to use a setup-job to consume env vars and
+provide an output that can be mapped to the arguments of the job.
+
+<details>
+<summary>Click here to see an example of this</summary>
+<code><pre>env:
+  AUTH_PROJECT_NUMBER: "123456789123"
+  SERVICE_ACCOUNT: sa-name@project-name-123.iam.gserviceaccount.com
+  PROJECT_ID: project-name-123
+jobs:
+  setup-env:
+    runs-on: ubuntu-latest
+    outputs:
+      auth_project_number: ${{ steps.set-output.outputs.auth_project_number }}
+      service_account: ${{ steps.set-output.outputs.service_account }}
+      project_id: ${{ steps.set-output.outputs.project_id }}
+    steps:
+      - name: set outputs with default values
+        id: set-output
+        run: |
+          echo "auth_project_number=${{ env.AUTH_PROJECT_NUMBER }}" >> $GITHUB_OUTPUT
+          echo "service_account=${{ env.SERVICE_ACCOUNT }}" >> $GITHUB_OUTPUT
+          echo "project_id=${{ env.PROJECT_ID }}" >> $GITHUB_OUTPUT
+  dev:
+    name: Deploy to dev
+    needs: setup-env
+    permissions:
+      id-token: write
+      contents: read
+      pull-requests: write
+      packages: write
+    uses: kartverket/github-workflows/.github/workflows/run-terraform.yml@v2.1
+    with:
+      runner: atkv1-dev
+      environment: dev
+      kubernetes_cluster: atkv1-dev
+      terraform_workspace: dev
+      terraform_option_1: -var-file=dev.tfvars -var=image=<registry>/<repository>:<tag> or <registry>/<repository>@<digest>
+      working_directory: terraform
+      auth_project_number: ${{ needs.setup-env.outputs.auth_project_number }}
+      service_account: ${{ needs.setup-env.outputs.service_account }}
+      project_id: ${{ needs.setup-env.outputs.project_id }}</pre></code>
+      image_url: <registry>/<repository>:<tag> or <registry>/<repository>@<digest> # the image created by the build job
+</details>
+<br />
+
+## Passing secrets to reusable workflows
+
+Secrets should be administered through vault and thus there is no explicit
+support for taking GitHub secrets into the action. GitHub will also stop you
+from sending secrets as arguments using the `with` argument, so this will not
+work either.
+
+Use the [vault_generic_secret](https://registry.terraform.io/providers/hashicorp/vault/latest/docs/data-sources/generic_secret)
+Terraform data source along with the `vault_role` argument to run-terraform to
+fetch secrets required at deploy-time.
+
+Each repo needs its own unique `vault_role`. Contact SKIP if you do not have
+this role.
+
+<br />
+
+# Troubleshooting
+See [TROUBLESHOOTING.md](https://github.com/kartverket/github-workflows/blob/main/TROUBLESHOOTING.md)
```

Comment:
https://stackoverflow.com/questions/7653483/github-relative-link-in-markdown-file

Bruke relativ lenke her istedet. Lenken funker nå ikke, og sender deg til `main` branch troubleshooting.